### PR TITLE
Implement TypeScript

### PR DESCRIPTION
**Pull request by fitzgeraldkd**
Migrated to TypeScript and made some slight changes to the README. I made a couple of minor improvements that were flagged now that TypeScript is implemented too, I'll flag those with comments 🚀 

Since this is TS now, the files in the `dist` directory are automatically generated from the `src` directory.

### CODE REVIEW

File: src/analytics.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ function getEventProperties(eventName, target) {
 
 // Assign a click event for any element with `data-event-name` set.
 // The element should also have the `data-event-label` and `data-event-block` custom attributes set.
-$('[data-event-name]').click(function() {
+$('[data-event-name]').on('click', function() {
```
Comment (fitzgeraldkd) [inline src/analytics.ts:110]: The `.click(...)` method is deprecated, so using `.on('click', ...)` instead.

File: src/player.ts [hunk 2]
```diff
@@ -132,17 +131,15 @@ function resetControllers() {
...
     // revised.
     const videos = $(videoToggle).parent().find('video')
     if (videos.length === 0) {
         Sentry.captureMessage('A video toggle button does not have an associated video.')
         return null
     } else if (videos.length > 1) {
-        Sentry.captureMessage('Multiple videos were found associated with a video toggle button. Only the first video will be controlled by the button.', {
-            level: 'info'
-        })
+        Sentry.captureMessage('Multiple videos were found associated with a video toggle button. Only the first video will be controlled by the button.', 'info')
```
Comment (fitzgeraldkd) [inline src/player.ts:142]: The warning level shouldn't have been wrapped in an object, it's just the second argument to `captureMessage`

File: src/player.ts [hunk 3]
```diff
@@ -153,14 +150,12 @@ function findAssociatedVideo(videoToggle) {
  * safely retrieved and the icon in the button can be changed.
  * Videos on the page will autoplay on loop, so the buttons control the mute state of the video element.
  */
-function toggleVideoMute() {
-    const videoToggle = this
```
Comment (fitzgeraldkd) [inline src/player.ts:157]: The recommended TypeScript/ESLint rules includes a rule to not allow aliasing `this`.

**Review by martingouy — APPROVED** [review]
Awesome, thanks for doing this! I didn't review lines 1 by 1, but I trust Typescript to find any code error haha.

The pre-commit hook is a very good call!
For later improvements
- https://typicode.github.io/husky/#/ -> Could help install the hook upon npm ci which could be handy
- Wondering what is the recommended way of handling files saved but not committed.  (to avoid having to think about stash)

File: src/analytics.ts (hunk 1 above)
Comment (MoncefB) [reply src/analytics.ts:110]: Thanks for the heads-up!

**Review by MoncefB — COMMENTED** [review]
Great initiative! And I'm glad the pre-commit hook ended up being a viable solution 👌

**Pull request merged**